ui/components/src/TimeRangeSelector/TimeRangeSelector.tsx

```diff
@@ -133,13 +143,15 @@ export function TimeRangeSelector(props: TimeRangeSelectorProps) {
               {item.display}
             </MenuItem>
           ))}
-          <MenuItem
-            key={timeOptions.length}
-            value={formatTimeRange(lastOption.value, timeZone)}
-            onClick={() => setShowCustomDateSelector(true)}
-          >
-            {lastOption.display}
-          </MenuItem>
+          {showCustomTimeRange && (
+            <MenuItem
+              key={timeOptions.length}
```

Comment:
key can be removed for this one

Reply:
I never understood why key is sometimes needed sometimes not

Reply:
Just removed it and nothing complains so ok ^^'